dynawo/sources/Models/Modelica/Dynawo/Electrical/Shunts/ShuntBWithSections.mo

```diff
@@ -0,0 +1,68 @@
+within Dynawo.Electrical.Shunts;
+
+/*
+* Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model ShuntBWithSections "Shunt element with voltage dependent reactive power and a variable susceptance given by a table and a section"
+
+  import Dynawo.Connectors;
+  import Dynawo.Types;
+  import Dynawo.Electrical.Controls.Basics.SwitchOff;
+  import Modelica;
+
+  extends SwitchOff.SwitchOffShunt;
+
+public
+
+  Connectors.ACPower terminal(V(re(start = u0Pu.re), im(start = u0Pu.im)), i(re(start = i0Pu.re), im(start = i0Pu.im))) "Connector used to connect the shunt to the grid";
+  Connectors.ZPin section(value(start = section0)) "section position of the shunt";
+
+  parameter Real section0 "Initial section of the shunt";
+  parameter String tableBPuName "Name of the table to calculate BPu from the section of the shunt";
+  parameter String tableBPuFile "File containing the table to calculate BPu from the section of the shunt";
+
+  Types.VoltageModulePu UPu(start = ComplexMath.'abs'(u0Pu)) "Voltage amplitude at shunt terminal in p.u (base UNom)";
+  Types.ActivePowerPu PPu(start = 0) "Active power at shunt terminal in p.u (base SnRef, receptor convention)";
+  Types.ReactivePowerPu QPu(start = s0Pu.im) "Reactive power at shunt terminal in p.u (base SnRef, receptor convention)";
+  Types.ComplexApparentPowerPu SPu(re(start = 0), im(start = s0Pu.im)) "Apparent power at shunt terminal in p.u (base SnRef, receptor convention)";
+  Types.PerUnit BPu(start = - s0Pu.im / ComplexMath.'abs'(u0Pu)^2) "Variable susceptance of the shunt in p.u (base SnRef, UNom)";
+  Modelica.Blocks.Tables.CombiTable1D tableBPu(tableOnFile = true, tableName = tableBPuName, fileName = tableBPuFile) "Table to get BPu from the section of the shunt";
```

Comment:
The reason why we do this is that we don't need to choose the size of the table in the .mo but directly in the .txt.
This allows us to cope with variable size tables (I suppose that all shunts don't have the same number of sections, which is quite difficult otherwise in Modelica I guess).
Another solution could be to choose a max size and use an oversized set of parameters directly in the .mo. We don't know yet which is best/worst in terms of performances (too much parameters or a .txt file to read).